src/com/ansorgit/plugins/bash/lang/lexer/bash.flex

```diff
@@ -320,7 +336,7 @@ Filedescriptor = "&" {IntegerLiteral} | "&-"
     }
 }
 
-<YYINITIAL, S_CASE> {
+<YYINITIAL, S_CASE, S_HEREDOC_EXPECTED> {
```

Comment:
I agree that this looks suspicious.
The reason is more or less a hack to keep the lexer in a non-0 state. See my note above `lexing when heredocs are expected is done in a non-0 state to force the highlighting to update after changes to the heredoc`.

When a heredoc start token is encountered then the lexer goes into the `S_HEREDOC_EXPECTED` state because it's only used to not do incremental lexing. This state has to match everything as before, that's why it's added to all the lists of states.

If you know a better way to get rid of incremental (highlighter) lexing please let me know.

Reply:
@mmazurkevich what do you think?

Reply:
The main problem is that this lexer has a lot of `YYINITIAL` blocks, that's why it's needed to put additional state into at least all of them.  This state `S_HEREDOC_EXPECTED` will end at the end of line where heredoc  marker appear. 
We named this state `HERE_DOC_PIPELINE` because pipeline can be after heredoc marker start.
@ignatov A lot of `S_HEREDOC_EXPECTED` simply because of a lot of `YYINITIAL` blocks, for this lexer it's ok there is no way to fix it for now.
@jansorg I simply add some review notes may be they will be helpful.